cpp/CMakeLists.txt

```diff
@@ -21,14 +21,16 @@ set(COMMON_INCLUDES ${PROJECT_SOURCE_DIR})
 include_directories(${EIGEN_INCLUDE_DIRS})
 include_directories(${COMMON_INCLUDES}) 
 
-# Set source files
-file(GLOB_RECURSE SOURCES RELATIVE ${CMAKE_SOURCE_DIR} src/*.cc)
-message(STATUS "Source files: " ${SOURCES})
 
 # Locate CUDA package 
 find_package(CUDA)
 if (CUDA_FOUND)
 
+  # Set source files
+  file(GLOB_RECURSE SOURCES RELATIVE ${CMAKE_SOURCE_DIR} src/*.cc src/core/gpu_util/*.cu)
```

Comment:
If I change the extension to .cc then the compiler does not know how to handle the cuda API calls in the definition